src/sfma/models/marginal.py

```diff
@@ -70,15 +72,23 @@ def remat(self) -> ndarray:
     def iemat(self) -> ndarray:
         return self.params["eta"].design_matrix_fe
 
-    def get_vars(self, x: ndarray) -> Tuple[ndarray]:
+    def get_vars(self, x: ndarray) -> Tuple[ndarray, ndarray, ndarray]:
         variables = np.split(x, np.cumsum([self.var_sizes])[:-1])
         beta = variables[0]
         gamma = np.sqrt(variables[1]**2)
         eta = np.sqrt(variables[2]**2)
         return beta, gamma, eta
 
+    @property
+    def w(self):
+        return self._w
+
+    @w.setter
+    def w(self, weights):
```

Comment:
To use take advantage of `setter` we could add checks of the weights, to enforce it between 0 and 1.
```
if any(map(lambda x: x < 0 or x > 1), weights):
    raise ValueError
```

Reply:
I like that! 5a28339